### PR TITLE
Hotfix bundle

### PR DESCRIPTION
**Pull request by tarcisiozf**
* Fix DMS and credentials update via settings 
* Debounce handling `auth.submit` event to avoid API rate errors (small nonce)
* Add coverage reporting
* Better error handling when validating api permissions
* Add more tests

### CODE REVIEW

**Review by vigan-abd — APPROVED** [review]
lgtm

**Review by avsek477 — CHANGES_REQUESTED** [review]
rest looks good to me

File: lib/ws_servers/api/start_connections.js [hunk 1]
```diff
@@ -6,35 +6,90 @@ const DmsRemoteControl = require('./dms_remote_control')
...
+   * @param args.restURL
+   * @param args.hostedURL
+   * @param args.dmsScope
+   * @returns {Promise<void>}
+   */
+  async start (args) {
+    await this._startDmsControl(args)
+
+    const { ws, apiKey, apiSecret } = args
+    const newCredentials = apiKey !== this.apiKey && apiSecret !== this.apiSecret
```
[the remark this answers is not in the document]
Comment (tarcisiozf) [reply lib/ws_servers/api/start_connections.js]: https://github.com/bitfinexcom/bfx-hf-server/pull/145/commits/fbddafaa3be591897db0423303bf218ec99b0533

**Review by avsek477 — APPROVED** [review]
lgtm

**Pull request merged**